### PR TITLE
[FLINK-19043][docs-zh] Translate the 'Logging' page of 'Debugging & Monitoring' into Chinese

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

*Translate the 'Logging' page of 'Debugging & Monitoring' into Chinese*

## Brief change log

*Translate the 'Logging' page of 'Debugging & Monitoring' into Chinese*
  - *The page url is : Logging https://ci.apache.org/projects/flink/flink-docs-release-1.11/monitoring/logging.html*
  - *The markdown file location is : flink/docs/monitoring/logging.zh.md*

## Verifying this change

*Translate the 'Logging' page of 'Debugging & Monitoring' into Chinese*

A pure translation work in the `Documentation` module.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable & docs

### CODE REVIEW

File: docs/monitoring/logging.zh.md [hunk 1]
```diff
@@ -100,15 +106,13 @@ import org.slf4j.Logger
...
-Using placeholders allows to avoid unnecessary string constructions in case that the logging level is set so high that the message would not be logged.
-The syntax of placeholders is the following:
+为了最大限度地利用 slf4j，建议使用其占位符机制。使用占位符可以避免不必要的字符串构造，以防日志级别设置得太高而不会记录消息。占位符的语法如下：
 
 {% highlight java %}
 LOG.info("This message contains {} placeholders. {}", 2, "Yippie");
 {% endhighlight %}
 
-Placeholders can also be used in conjunction with exceptions which shall be logged.
+占位符也可以与应记录的异常一起使用。
```
Comment (RocMarshal) [inline docs/monitoring/logging.zh.md:115]:
```suggestion
占位符也可以和要记录的异常一起使用。
```

**Comment by RocMarshal** [comment]
Hi, @klion26 
Could you help me to review this PR if you have free time?
Thank you.

File: docs/monitoring/logging.zh.md [hunk 2]
```diff
@@ -23,47 +23,51 @@ specific language governing permissions and limitations
...
+- `log4j.properties`：作为 JobManager/TaskManager 日志配置使用（standalone 和 YARN 两种模式下皆使用）
 
-Flink ships with the [Log4j API bridge](https://logging.apache.org/log4j/log4j-2.2/log4j-1.2-api/index.html), allowing existing applications that work against Log4j1 classes to continue working.
+<a name="compatibility-with-log4j1"></a>
 
-If you have custom Log4j1 properties files or code that relies on Log4j1, please check out the official Log4j [compatibility](https://logging.apache.org/log4j/2.x/manual/compatibility.html) and [migration](https://logging.apache.org/log4j/2.x/manual/migration.html) guides.
+### 与 Log4j1 的兼容性
 
-## Configuring Log4j1
+Flink 附带了 [Log4j API bridge](https://logging.apache.org/log4j/log4j-2.2/log4j-1.2-api/index.html)，使得对 Log4j1 工作的现有应用程序继续工作。
```
Comment (klion26) [inline docs/monitoring/logging.zh.md:47]: `使得对 Log4j1 工作的现有应用程序继续工作` 这句话意思能清楚，读起来有一点拗口，能否改进一下呢？比如 `使得现在作业能够继续使用 log4j1 的接口` （类似的描述）

File: docs/monitoring/logging.zh.md [hunk 3]
```diff
@@ -23,47 +23,51 @@ specific language governing permissions and limitations
...
 
-To use Flink with Log4j1 you must ensure that:
-- `org.apache.logging.log4j:log4j-core`, `org.apache.logging.log4j:log4j-slf4j-impl` and `org.apache.logging.log4j:log4j-1.2-api` are not on the classpath,
-- `log4j:log4j`, `org.slf4j:slf4j-log4j12`, `org.apache.logging.log4j:log4j-to-slf4j` and `org.apache.logging.log4j:log4j-api` are on the classpath.
+如果你有基于 Log4j1 的自定义配置文件或代码，请查看官方 Log4j [兼容性](https://logging.apache.org/log4j/2.x/manual/compatibility.html)和[迁移](https://logging.apache.org/log4j/2.x/manual/migration.html)指南。
```
Comment (klion26) [inline docs/monitoring/logging.zh.md:49]: `Log4j 的` 会更好一些吗？

File: docs/monitoring/logging.zh.md [hunk 4]
```diff
@@ -23,47 +23,51 @@ specific language governing permissions and limitations
...
+要将 Flink 与 Log4j1 一起使用，必须确保：
+- Classpath 中不存在 `org.apache.logging.log4j:log4j-core`，`org.apache.logging.log4j:log4j-slf4j-impl` 和 `org.apache.logging.log4j:log4j-1.2-api`，
+- 且 Classpath 中存在 `log4j:log4j`，`org.slf4j:slf4j-log4j12`，`org.apache.logging.log4j:log4j-to-slf4j` 和 `org.apache.logging.log4j:log4j-api`。
 
-For users and developers alike it is important to control the logging framework.
-The configuration of the logging framework is exclusively done by configuration files.
-The configuration file either has to be specified by setting the environment property `-Dlogback.configurationFile=<file>` or by putting `logback.xml` in the classpath.
-The `conf` directory contains a `logback.xml` file which can be modified and is used if Flink is started outside of an IDE and with the provided starting scripts.
-The provided `logback.xml` has the following form:
+在 IDE 中，这意味着你必须替换在 pom 文件中定义的依赖项，并尽可能在传递依赖于它们的依赖项上添加排除项。
```
Comment (klion26) [inline docs/monitoring/logging.zh.md:59]: `并尽可能在传递依赖于它们的依赖项上添加排除项` 这句话能否再优化一下呢？读起来有点拗口

File: docs/monitoring/logging.zh.md [hunk 5]
```diff
@@ -81,17 +85,19 @@ The provided `logback.xml` has the following form:
...
-For further information on configuring logback see [LOGback's manual](http://logback.qos.ch/manual/configuration.html).
+有关配置日志的更多信息，请参见 [LOGback 手册](http://logback.qos.ch/manual/configuration.html)。
+
+<a name="best-practices-for-developers"></a>
 
-## Best practices for developers
+## 开发人员的最佳实践
 
-The loggers using slf4j are created by calling
+Slf4j 的 loggers 通过调用来创建
```
Comment (klion26) [inline docs/monitoring/logging.zh.md:100]: 这句话读起来感觉有点怪怪的，这句话没有和下面的代码有联系，原文中这句话会和下面的代码有联系

File: docs/monitoring/logging.zh.md [hunk 6]
```diff
@@ -100,15 +106,13 @@ import org.slf4j.Logger
 Logger LOG = LoggerFactory.getLogger(Foobar.class)
 {% endhighlight %}
 
-In order to benefit most from slf4j, it is recommended to use its placeholder mechanism.
-Using placeholders allows to avoid unnecessary string constructions in case that the logging level is set so high that the message would not be logged.
-The syntax of placeholders is the following:
+为了最大限度地利用 slf4j，建议使用其占位符机制。使用占位符可以避免不必要的字符串构造，以防日志级别设置得太高而不会记录消息。占位符的语法如下：
```
Comment (klion26) [inline docs/monitoring/logging.zh.md:109]: beneift 是不是翻译成受益会更好一些呢？（整体意思或语句可能需要调整）后面是解释为什么使用占位符会更受益
后面一句话的意思也需要修改一下，如果我理解没错的话，是说使用占位符可以避免一些不必要的字符串构建（比如 log 设置为 WARN，那么使用占位符的话，DEBUG 和 INFO 的 log 就不会有字符串构造了）

**Review by klion26 — COMMENTED** [review]
@RocMarshal thanks for your contribution, left some comments, please let me know what do you think ,thanks.

File: docs/monitoring/logging.zh.md (hunk 3 above)
Comment (RocMarshal) [inline docs/monitoring/logging.zh.md:49]:
```suggestion
如果你有基于 Log4j 的自定义配置文件或代码，请查看官方 Log4j [兼容性](https://logging.apache.org/log4j/2.x/manual/compatibility.html)和[迁移](https://logging.apache.org/log4j/2.x/manual/migration.html)指南。
```

File: docs/monitoring/logging.zh.md (hunk 5 above)
Comment (RocMarshal) [inline docs/monitoring/logging.zh.md:100]:
```suggestion
Slf4j 的 loggers 通过调用 `LoggerFactory` 的 `getLogger()` 方法创建
```

File: docs/monitoring/logging.zh.md (hunk 4 above)
Comment (RocMarshal) [reply docs/monitoring/logging.zh.md:59]:
```suggestion
在 IDE 中使用 log4j1，你必须在 pom 文件中使用上述 `Classpath 中存在的 jars` 依赖项替换 `Classpath 中不存在的 jars` 依赖项，并尽可能在传递依赖于 `Classpath 中不存在的 jars` 的依赖项上添加排除 `Classpath 中不存在的 jars` 配置。
```

File: docs/monitoring/logging.zh.md (hunk 2 above)
Comment (RocMarshal) [inline docs/monitoring/logging.zh.md:47]:
```suggestion
Flink 附带了 [Log4j API bridge](https://logging.apache.org/log4j/log4j-2.2/log4j-1.2-api/index.html)，使得现有作业能够继续使用 log4j1 的接口。
```

**Review by klion26 — APPROVED** [review]
@RocMarshal thanks for addressing the comments. LGTM
ping @wuchong @rmetzger

**Comment by RocMarshal** [comment]
LGTM.
ping @wuchong , @dianfu

**Comment by rmetzger** [comment]
Thanks for the review! I'll merge the change now!

**Pull request closed without merging**